### PR TITLE
chore: renew verify conextual structured script

### PR DESCRIPTION
**Pull request by googs1025**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes # part of https://github.com/kubernetes-sigs/scheduler-plugins/issues/743

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/sysched/sysched.go [hunk 1]
```diff
@@ -134,6 +134,9 @@ func (sc *SySched) getSyscalls(logger klog.Logger, pod *v1.Pod) sets.Set[string]
 			if len(ns) > 0 && len(name) > 0 {
 				syscalls, err := sc.readSPOProfileCR(name, ns)
 				if err != nil {
+					klog.ErrorS(err, "aaaa")
```
Comment (googs1025) [inline pkg/sysched/sysched.go:137]: Just to test whether the script works as expected

**Comment by googs1025** [comment]
use this as example: https://github.com/kubernetes-sigs/scheduler-plugins/pull/831#discussion_r1835238891
ci workflow works fine


https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_scheduler-plugins/831/pull-scheduler-plugins-verify/1855090645207617536

**Comment by googs1025** [comment]
When this script is merged, we can only use conextual logs like `logger := klog.FromContext(ctx)`. If using methods like klog.InfoS, we will get an error in ci workflow.

File: hack/verify-structured-logging.sh [hunk 2]
```diff
@@ -46,9 +50,35 @@ done < <(echo "${all_packages[@]}" | tr " " "\n")
...
+  ret=1
+fi
 
 if [ $ret -eq 0 ]; then
   echo "Structured logging static check passed on all packages :)"
-fi
+else
+  echo "Structured logging static check failed. Please fix the issues."
+  exit 1
+fi
```
Comment (Huang-Wei) [inline hack/verify-structured-logging.sh:84]: nit: keep the newline :)

File: hack/verify-structured-logging.sh [hunk 3]
```diff
@@ -46,9 +50,35 @@ done < <(echo "${all_packages[@]}" | tr " " "\n")
...
-GOOS=linux    logcheck "${packages[@]}" || ret=$?
-GOOS=windows  logcheck "${packages[@]}" || ret=$?
+
+# Function to run logcheck and capture output
+run_logcheck() {
+  local os="$1"
+  shift
+  local output
+  output=$(GOOS="$os" logcheck "$@" 2>&1)
+  echo "$output"
```
Comment (Huang-Wei) [inline hack/verify-structured-logging.sh:60]: My 2 cents:

- remove this line here as for current code base, it reports a lot of violations. I may choose to resolve them in phases
- in Phase 1: treat violations with string "unstructured logging function" as fatal error, dump the problematic entries under L62 
- in Phase 2: fix all violations, and hence don't need to filter out "unstructured logging function"

File: hack/verify-structured-logging.sh [hunk 4]
```diff
@@ -46,9 +50,35 @@ done < <(echo "${all_packages[@]}" | tr " " "\n")
...
+echo "Running logcheck for Linux..."
+if ! run_logcheck linux "${packages[@]}"; then
+  ret=1
+fi
+
+# Run logcheck for Windows
+echo "Running logcheck for Windows..."
+if ! run_logcheck windows "${packages[@]}"; then
+  ret=1
+fi
```
Comment (Huang-Wei) [inline hack/verify-structured-logging.sh:77]: I think previous code to run against **both** linux and windows is a fault. We can remove running against windows.

File: hack/verify-structured-logging.sh [hunk 5]
```diff
@@ -46,9 +50,35 @@ done < <(echo "${all_packages[@]}" | tr " " "\n")
...
-GOOS=windows  logcheck "${packages[@]}" || ret=$?
+
+# Function to run logcheck and capture output
+run_logcheck() {
+  local os="$1"
+  shift
+  local output
+  output=$(GOOS="$os" logcheck "$@" 2>&1)
+  echo "$output"
+  if echo "$output" | grep -q 'unstructured logging function'; then
```
[the remark this answers is not in the document]
Comment (Huang-Wei) [reply hack/verify-structured-logging.sh:61]: yup, we can target it as the goal of phase 1.

File: hack/verify-structured-logging.sh (hunk 4 above)
Comment (googs1025) [reply hack/verify-structured-logging.sh]: done

File: hack/verify-structured-logging.sh (hunk 3 above)
Comment (googs1025) [reply hack/verify-structured-logging.sh]: removed.

In addition, I added todo items to record these things 🤔

File: hack/verify-structured-logging.sh [hunk 6]
```diff
@@ -46,9 +50,31 @@ done < <(echo "${all_packages[@]}" | tr " " "\n")
...
+run_logcheck() {
+  local os="$1"
+  shift
+  local output
+  output=$(GOOS="$os" logcheck "$@" 2>&1)
+  # TODO: Add more checks as needed
+  # Currently, we only check for unstructured logging
+  # more detail: https://github.com/kubernetes-sigs/scheduler-plugins/pull/831#discussion_r1835553079
+  if echo "$output" | grep -q 'unstructured logging function'; then
+    echo "Error: Unstructured logging detected. Please fix the issues."
```
Comment (Huang-Wei) [inline hack/verify-structured-logging.sh:64]: I think we need to use a variable in L63 to hold the errorous entries, so as to dump below L64, right? (pseudo code below, please adjust/evalaute)

```bash
errors=$(echo "$output" | grep -q 'unstructured logging function')
if [[ $? -ne 0 ]]; then
   echo "Error blabla"
   echo "$errors"
```

File: hack/verify-structured-logging.sh [hunk 7]
```diff
@@ -26,9 +26,13 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 source "${KUBE_ROOT}/hack/lib/util.sh"
 
+export GOPATH=${GOPATH:-$HOME/go}
+export GOBIN=${GOBIN:-$GOPATH/bin}
+export PATH=$GOBIN:$PATH
```
Comment (Huang-Wei) [inline hack/verify-structured-logging.sh:31]: do we need this?

File: hack/verify-structured-logging.sh [hunk 8]
```diff
@@ -26,9 +26,13 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
...
+
 kube::golang::verify_go_version
 
-GO111MODULE=on go install k8s.io/klog/hack/tools/logcheck@latest
+GO111MODULE=on go install sigs.k8s.io/logtools/logcheck@latest
```
Comment (Huang-Wei) [inline hack/verify-structured-logging.sh:35]: we can stick to k8s.io/klog in this PR. I have a WIP #832 taking care of this.

File: hack/verify-structured-logging.sh [hunk 9]
```diff
@@ -46,9 +46,33 @@ done < <(echo "${all_packages[@]}" | tr " " "\n")
...
+run_logcheck() {
+  local os="$1"
+  shift
+  local output
+  output=$(GOOS="$os" logcheck "$@" 2>&1)
+  # TODO: Add more checks as needed
+  # Currently, we only check for unstructured logging
+  # more detail: https://github.com/kubernetes-sigs/scheduler-plugins/pull/831#discussion_r1835553079
+  errors=$(echo "$output" | grep 'unstructured logging function')
+  if [[ $? -eq 0 ]]; then
```
Comment (Huang-Wei) [inline hack/verify-structured-logging.sh:60]: nope, it should ensure the return code of previous command not equal 0 - i.e., `-ne`

Comment (googs1025) [reply hack/verify-structured-logging.sh]: ok got this

Comment (Huang-Wei) [reply hack/verify-structured-logging.sh]: sry my bad 🤦

`grep` returns 0 to indicate there is a hit - in our case, it's an error. So your previous version is correct.

Comment (googs1025) [reply hack/verify-structured-logging.sh:60]: ok

File: hack/verify-structured-logging.sh (hunk 7 above)
Comment (googs1025) [reply hack/verify-structured-logging.sh]: removed

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**